models/Document.php

```diff
@@ -464,7 +464,7 @@ static function (array $doc) use ($oldPath, $newPath): array {
             // refresh the inherited properties and update dependencies of each children
             if ($differentOldPath && isset($updatedChildren) && is_array($updatedChildren)) {
                 foreach ($updatedChildren as $updatedDocument) {
-                    $updatedDocument = self::getById($updatedDocument['id'], true);
+                    $updatedDocument = Document::getById($updatedDocument['id'], ['force' => true]);
```

Comment:
Should we put a check before calling `renewInheritedProperties`? as `getById` can return null in cases where the type is not matched. maybe an edge case but doesn't harm ...right?

Reply:
we could then type hint the $element param in `updateDependendencies` and no need to do match explicitly.

Reply:
sounds good, this commit would do https://github.com/pimcore/pimcore/pull/15420/commits/f5b88407e78491d5c88c02021655b640a33edaf8